agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_unsupported.go

```diff
@@ -0,0 +1,183 @@
+	return false
+}
+
+// TerminalStatus returns the last transition state of fsxwindowsfileserver resource
+func (fv *FSxWindowsFileServerResource) TerminalStatus() resourcestatus.ResourceStatus {
+	return resourcestatus.ResourceStatusNone
+}
+
+// NextKnownState returns the state that the resource should
+// progress to based on its `KnownState`.
+func (fv *FSxWindowsFileServerResource) NextKnownState() resourcestatus.ResourceStatus {
+	return resourcestatus.ResourceStatusNone
+}
+
+// ApplyTransition calls the function required to move to the specified status
+func (fv *FSxWindowsFileServerResource) ApplyTransition(nextState resourcestatus.ResourceStatus) error {
+	return errors.New("not implemented")
+}
+
+// SteadyState returns the transition state of the resource defined as "ready"
+func (fv *FSxWindowsFileServerResource) SteadyState() resourcestatus.ResourceStatus {
+	return resourcestatus.ResourceStatusNone
+}
+
+// SetKnownStatus safely sets the currently known status of the resource
+func (fv *FSxWindowsFileServerResource) SetKnownStatus(status resourcestatus.ResourceStatus) {
+}
+
+// SetAppliedStatus sets the applied status of resource and returns whether
+// the resource is already in a transition
+func (fv *FSxWindowsFileServerResource) SetAppliedStatus(status resourcestatus.ResourceStatus) bool {
+	return false
+}
+
+// GetKnownStatus safely returns the currently known status of the task
+func (fv *FSxWindowsFileServerResource) GetKnownStatus() resourcestatus.ResourceStatus {
+	return resourcestatus.ResourceStatusNone
+}
+
+// StatusString returns the string of the fsxwindowsfileserver resource status
+func (fv *FSxWindowsFileServerResource) StatusString(status resourcestatus.ResourceStatus) string {
+	return "undefined"
+}
+
+// SetCreatedAt sets the timestamp for resource's creation time
+func (fv *FSxWindowsFileServerResource) SetCreatedAt(createdAt time.Time) {
+}
+
+// GetCreatedAt sets the timestamp for resource's creation time
+func (fv *FSxWindowsFileServerResource) GetCreatedAt() time.Time {
+	return time.Time{}
+}
+
+// Source returns the host path of the fsxwindowsfileserver resource which is used as the source of the volume mount
+func (cfg *FSxWindowsFileServerVolumeConfig) Source() string {
+	return "undefined"
+}
+
+// GetName safely returns the name of the resource
+func (fv *FSxWindowsFileServerResource) GetName() string {
+	return "undefined"
+}
+
+// Create is used to create all the fsxwindowsfileserver resources for a given task
+func (fv *FSxWindowsFileServerResource) Create() error {
+	return errors.New("not implemented")
+}
+
+// Cleanup removes the fsxwindowsfileserver resources created for the task
+func (fv *FSxWindowsFileServerResource) Cleanup() error {
+	return errors.New("not implemented")
+}
+
+// MarshalJSON serialises the FSxWindowsFileServerResourceJSON struct to JSON
+func (fv *FSxWindowsFileServerResource) MarshalJSON() ([]byte, error) {
+	return nil, errors.New("not implemented")
```

Comment:
please use consistent error messages for unsupported platforms

Reply:
will do